src/reducers/reducer-expenses.js

```diff
@@ -8,14 +8,12 @@ import moment from 'moment';
 // type checking
 const validateCreateExpense = (data) => (
   data.hasOwnProperty('name') &&
-    data.hasOwnProperty('cost') &&
-    data.hasOwnProperty('type')
+    data.hasOwnProperty('cost')
 );
 
 const validateEditExpense = (data) => (
   data.hasOwnProperty('name') ||
-    data.hasOwnProperty('cost') ||
-    data.hasOwnProperty('type')
+    data.hasOwnProperty('cost')
 );
 
 const checkExpenseExists = (expenseId, state) => (
```

Comment:
The name implies a boolean response but the function returns an object. `retrieveExistingExpense` would be better